### PR TITLE
CONN-2269 view chain-of-trust

### PR DESCRIPTION
**Pull request by trantang**
create a view for chain-of-trust
calling the web-service and getting the list of the chain

### CODE REVIEW

File: Product/Production/Adapters/General/CONNECTAdminGUI/src/main/webapp/certificateManager.xhtml [hunk 1]
```diff
@@ -284,25 +154,185 @@
...
+                                                                styleClass="ui-confirmdialog-yes" icon="ui-icon-check"
+                                                                action="#{certificateBean.refreshCacheForTrustStore}"
+                                                                update=":certTabView"
+                                                                oncomplete="PF('certRefreshConfirm').hide();" />
+                                                            <p:commandButton value="No" styleClass="ui-confirmdialog-no"
+                                                                icon="ui-icon-close"
+                                                                oncomplete="PF('certRefreshConfirm').hide();" />
+                                                        </p:confirmDialog>
+                                                        <h:outputText value=" &nbsp;&nbsp;" />
+                                                        <p:commandButton id="btnViewChainOfTrust" value="Chain"
```
Comment (TabassumJafri) [inline Product/Production/Adapters/General/CONNECTAdminGUI/src/main/webapp/certificateManager.xhtml]: Rename the button to View Chain Of Trust

File: Product/Production/Adapters/General/CONNECTAdminGUI/src/main/webapp/dlgCertMgrChain.xhtml [hunk 2]
```diff
@@ -0,0 +1,53 @@
+<ui:composition xmlns:h="http://java.sun.com/jsf/html" xmlns:ui="http://java.sun.com/jsf/facelets"
+    xmlns:p="http://primefaces.org/ui" xmlns="http://www.w3.org/1999/xhtml" xmlns:f="http://java.sun.com/jsf/core">
+
+    <div>
+        <p:dialog id="dlgChainOfTrust" header="Dialog Chain-of-Trust" widgetVar="wgvDlgChainOfTrust" modal="true"
```
Comment (TabassumJafri) [inline Product/Production/Adapters/General/CONNECTAdminGUI/src/main/webapp/dlgCertMgrChain.xhtml]: header should read as Chain of Trust view

File: Product/Production/Adapters/General/CONNECTAdminGUI/src/main/java/gov/hhs/fha/nhinc/admingui/managed/CertficateBean.java [hunk 3]
```diff
@@ -460,4 +469,29 @@ public void setRefreshCache(boolean refreshCache) {
         this.refreshCache = refreshCache;
     }
 
+    public void viewChainOfTrust() {
+        chainOfTrust = new HashMap<>();
+        selectedCertsChain = new ArrayList<>();
+
+        if (null == selectedTSCertificate) {
+            HelperUtil.addMessageError(TRUST_STORE_MSG, "Please choose a certificate to view-chain.");
```
Comment (plobre) [inline Product/Production/Adapters/General/CONNECTAdminGUI/src/main/java/gov/hhs/fha/nhinc/admingui/managed/CertficateBean.java]: remove the - in "view-chain"

File: Product/Production/Gateway/AdminGUIWebservices/src/main/java/gov/hhs/fha/nhinc/configadmin/ConfigAdmin.java [hunk 4]
```diff
@@ -273,4 +277,60 @@ private static boolean updateTrustStoreAlias(CertificateManager certManager,
...
+     * @see
+     * gov.hhs.fha.nhinc.configadmin.EntityConfigAdminPortType#listChainOfTrust(gov.hhs.fha.nhinc.common.configadmin.
+     * ListChainOfTrustRequestMessageType)
+     */
+    @Override
+    public ListCertificatesResponseMessageType listChainOfTrust(ListChainOfTrustRequestMessageType request) {
+        LOG.debug("listChainOfTrust--begin");
+        String alias = request.getAlias();
+        if(StringUtils.isBlank(alias)){
+            return buildListCertificatesResponseMessageType(false, "Certificate Alias is required for chain-of-trust");
```
Comment (plobre) [inline Product/Production/Gateway/AdminGUIWebservices/src/main/java/gov/hhs/fha/nhinc/configadmin/ConfigAdmin.java]: remove -'s

File: Product/Production/Gateway/AdminGUIWebservices/src/main/java/gov/hhs/fha/nhinc/configadmin/ConfigAdmin.java [hunk 5]
```diff
@@ -273,4 +277,60 @@ private static boolean updateTrustStoreAlias(CertificateManager certManager,
...
+        }
+
+        try {
+            ListCertificatesResponseMessageType response = new ListCertificatesResponseMessageType();
+            response.getCertList().addAll(buildChain(alias, getTrustStore()));
+            return response;
+        } catch (KeyStoreException | CertificateEncodingException ex) {
+            LOG.error("unable to build chain-of-trust: {}", ex.getMessage(), ex);
+            return buildListCertificatesResponseMessageType(false, "ERROR: Unable to build the chain-of-trust.");
```
Comment (plobre) [inline Product/Production/Gateway/AdminGUIWebservices/src/main/java/gov/hhs/fha/nhinc/configadmin/ConfigAdmin.java]: remove -'s.

Also why does this one have "ERROR" in it but a few lines (on 293) up we dont?

**Review by TabassumJafri — CHANGES_REQUESTED** [review]

**Review by plobre — APPROVED** [review]

**Review by TabassumJafri — APPROVED** [review]

**Review by vdmehta06 — APPROVED** [review]

**Pull request merged**